src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -338,8 +338,8 @@ private ImmutableArray<LSP.Diagnostic> ConvertDiagnostic(IDiagnosticSource diagn
 
                 // Defines an identifier used by the client for merging diagnostics across projects.
                 // We want diagnostics to be merged from separate projects if they have the same code, filepath, range, and message.
-                var filePath = diagnosticData.DataLocation.GetFilePath();
-                diagnostic.Identifier = (diagnostic.Code, filePath, diagnostic.Range, diagnostic.Message).GetHashCode().ToString();
+                diagnostic.Identifier = (diagnostic.Code, diagnosticData.DataLocation.OriginalFileSpan.Path, diagnostic.Range, diagnostic.Message)
+                    .GetHashCode().ToString();
```

Comment:
i switched this to just use OriginalFilePath.  This was due to the fact that the rest of hte code here only accesses OriginalFileSpan and never touches MappedFileSpan (presumably because razor handles those themselves and we are not supposed to).